### DISCUSSION TITLE
Merge IndexHNSW

### DISCUSSION THREAD

**Discussion by codeworse** [facebookresearch/faiss#4096] [Q&A]
Is there an efficient way to merge IndexHNSW? It seems odd to me that there is no implementation of `merge_from` in IndexHNSWFlat/PQ and etc. It could be easily added by doing `storage->merge_from(other.storage)` inside(and in case of nullptr storage throw an exception)

**Comment by mdouze** [comment]
Stitching graph indices is not trivial without incurring an accuracy loss or memory overhead. 
See an overview here
https://dl.acm.org/doi/pdf/10.1145/3627535.3638475
